libs/wingsdk/test/core/plugin-manager.test.ts

```diff
@@ -0,0 +1,148 @@
+import fs from "fs";
+import { join } from "path";
+import * as cdktf from "cdktf";
+import { PluginManager } from "../../src/core/plugin-manager";
+import * as tfaws from "../../src/target-tf-aws";
+import { mkdtemp } from "../../src/util";
+import { tfResourcesOfCount } from "../util";
+
+const PLUGIN_CODE = `
+var s3_bucket = require("@cdktf/provider-aws/lib/s3-bucket");
+
+exports.preSynth = function(app) {
+  new s3_bucket.S3Bucket(app, "PluginBucket", {bucket: "plugin-bucket"})  
+};
+
+exports.postSynth = function(config) {
+  //TODO: Not sure why I needed to do a JSON.parse here
```

Comment:
This feels like something we need to figure out because that's not the desired behavior